src/Interpreters/InterpreterGrantQuery.cpp

```diff
@@ -33,7 +33,12 @@ namespace
             if (query.is_revoke)
                 grantee.access.revoke(query.access_rights_elements);
             else
+            {
+                if (query.is_replace)
+                    grantee.access = {};
```

Comment:
`GRANT REPLACE USAGE ON *.* TO user1` should work too but it wouldn't because `query.access_rights_elements.empty() == true` in this case.

Reply:
@vitlibar what is `USAGE` privilage? there's no document on it

Reply:
forget it, I've found it.

Reply:
`USAGE` means no privileges, it's the same as `NONE`.

Reply:
ok, fixed, PTAL